nats-base-client/service.ts

```diff
@@ -421,16 +447,29 @@ export class ServiceImpl extends QueuedIteratorImpl<ServiceMsg>
     return h;
   }
 
-  countError(err: Error): void {
-    this._lastException = err;
-    this._stats.num_errors++;
-    this._stats.last_error = err.message;
+  countError(err: Error, name?: string): void {
```

Comment:
We should agree on how stats are handled, counted and returned.
Let's have a schema definition for all those responses. This will clear out any doubts and incosistencies between service implementations.

Again: lack of "root" endpoint will simplify also our internal code around handling stats.

Reply:
each of the endpoints tracks the same stats as the original.

Reply:
See https://github.com/nats-io/nats-architecture-and-design/pull/186#discussion_r1063024372 for the stats